### PR TITLE
:bug: Source ZohoCRM: Update API version, default optional fields to None, add `fields` request param

### LINKED ISSUE #27422: Source ZohoCRM: API Version is out of date and the optional fields are not defaulted to None
### Connector Name

source-zoho-crm

### Connector Version

0.1.2

### What step the error happened?

None

### Revelant information

For a specific customer account we were using, they were not seeing the Deals stream.
After investigating we found that the fields meta data coming back from the Deals stream was missing the "json_type" and "length" property in some of the data. This was causing the deals stream to not be added to the list of streams in the Zoho Streams Factory.
"json_type" is a required type thus this makes sense but "length" is not required thus it shouldn't cause an exit on that.
Upon further inspection we found that even though length is set to Optional, it was not being defaulted to None thus was actually a required field according to this stackoverflow post https://stackoverflow.com/questions/70809438/python-dataclasses-with-optional-attributes. 

The reason the "json_type" field was missing is unknown, it would be an issue on Zoho's API. But we found upgrading the api version in the connector from v2 to v4 fixes this problem and all the required fields are present.

Thus this simply requires 2 small fixes:
1. Updating v2 to v4 for the api version being used in the ZohoCRM connector
2. Have the optional fields in the FieldMeta class default to None

I will post a PR for this issue shortly.

### Relevant log output

_No response_

### Contribute

- [x] Yes, I want to contribute

PR for the fix found here:
https://github.com/airbytehq/airbyte/pull/27423

### PR DESCRIPTION
**Pull request by pabloescoder**
## What
Resolves #27422 
- Connector now uses API v4 instead of v2
- Made the optional fields in the FieldMeta class default to None
- Added the `fields` query parameter to the request url
- Updated unit tests to check for API v4 instead of v2

## How
- Added a variable to keep track of the API version being used, and modified the path functions to use that variable in the request URL
- Set default value of None for fields having type Optional in types.py
- Used the request_params method to add a query parameter for `fields` that includes upto 50 fields, rest are dropped. (ZohoCRM allows a max of 50 parameters/fields per query)

## Recommended reading order
1. `airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/types.py`
2. `airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/api.py`
3. `airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py`

## 🚨 User Impact 🚨
No breaking changes.

## Pre-merge Actions
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [X] Unit & integration tests added

</details>

image
All 65 unit tests passing.

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Thanks for the contribution 🚀 . I added it to the team backlog and we're planning to start next week the review process.

**Comment by pabloescoder** [comment]
> Thanks for the contribution 🚀 . I added it to the team backlog and we're planning to start next week the review process.
> 

That sounds great, thanks!

**Comment by marcosmarxm** [comment]
I made a quick review asking to update a few files this will help us during the code review. Sorry the long delay to get into your contribution. Probably this week or next one the team will start the process. Are you able to run the tests in your local setup? It can speed up a lot the process if you share the integration tests results

File: airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py [hunk 1]
```diff
@@ -37,14 +37,21 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
     def request_params(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
     ) -> MutableMapping[str, Any]:
-        return next_page_token or {}
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        if self.module:
+            if len(self.module.fields) > 50:
+                self.logger.warning("ZohoCRM only allows a max of 50 parameters per query, dropping all fields after 50")
+            params["fields"] = ",".join([field.api_name for field in self.module.fields][:50])
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py:44]: need describe this logic in docs file below

**Review by lazebnyi — APPROVED** [review]
LGTM, minor comments to add details in docs

File: airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py (hunk 1 above)
Comment (pabloescoder) [reply airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py:44]: Hey @lazebnyi. Sorry for the delay, where exactly do I need to describe the logic in the docs? In the Notes section, the changelog description or some other place?

Comment (pabloescoder) [reply airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/streams.py]: @lazebnyi I've added a comment in the streams.py file as well as described why fields after 50 are dropped in the notes section of the docs. Do let me know if there are any further changes.

**Review by lazebnyi — APPROVED** [review]

**Comment by quangdvn-sun** [comment]
@marcosmarxm Might I ask why this PR has not been approved to be released yet ?

**Comment by marcosmarxm** [comment]
> @marcosmarxm Might I ask why this PR has not been approved to be released yet ?

Integration tests are working and the team has quite a large backlog to review and merge.

**Comment by quangdvn-sun** [comment]
Thanks for your information. I understand now 🙇

**Comment by amrael** [comment]
Has there been any development on this PR? We remain incapable of extracting data from the necessary streams due to the issue.

**Comment by marcosmarxm** [comment]
Hello @pabloescoder, I've chosen to close this contribution. Currently, we are working on migrating the Zoho CRM to a low-code framework, which includes an update to the new API. With a small team, it's impossible for us to review the contribution at this time.

**Comment by sbrudz** [comment]
> Currently, we are working on migrating the Zoho CRM to a low-code framework, which includes an update to the new API.

@marcosmarxm Do you have a timeline for the migration to the low-code framework?  Also, is it something that you'd accept community contributions on?

**Pull request closed without merging**